scene/2d/tile_map_layer.cpp

```diff
@@ -1085,56 +1098,112 @@ void TileMapLayer::_physics_draw_quadrant_debug(const RID &p_canvas_item, DebugQ
 						debug_collision_color.get_v() + rand.random(-1.0, 1.0) * 0.1,
 						debug_collision_color.a);
 
-				int shape_count = ps->body_get_shape_count(body);
 				for (int shape_index = 0; shape_index < shape_count; shape_index++) {
 					const RID &shape = ps->body_get_shape(body, shape_index);
 					const Transform2D &shape_xform = ps->body_get_shape_transform(body, shape_index);
 					const PhysicsServer2D::ShapeType &type = ps->shape_get_type(shape);
 
 					if (type == PhysicsServer2D::SHAPE_CONVEX_POLYGON) {
 						PackedVector2Array outline = ps->shape_get_data(shape);
+						const int outline_size = outline.size();
+						if (outline_size < 3) {
+							continue;
+						}
 
-						PackedVector2Array vertex_array;
-						vertex_array.resize(outline.size() + 1);
+						const Color outline_random_variation_color = random_variation_color.lightened(0.2);
+						const Transform2D outline_xform = body_to_quadrant * shape_xform;
 
-						PackedColorArray face_color_array;
-						face_color_array.resize(outline.size() + 1);
-						PackedInt32Array face_index_array;
-						face_index_array.resize((outline.size() - 2) * 3);
+						// Adds debug mesh lines.
 
-						PackedColorArray line_color_array;
-						line_color_array.resize(outline.size() + 1);
+						for (int i = 0; i < outline_size; i++) {
+							const Vector2 vertex1 = outline_xform.xform(outline[i % outline_size]);
+							const Vector2 vertex2 = outline_xform.xform(outline[(i + 1) % outline_size]);
 
-						for (int i = 0; i < outline.size() + 1; i++) {
-							Vector2 vertex = (body_to_quadrant * shape_xform).xform(outline[i % outline.size()]);
-							vertex_array.write[i] = vertex;
-							face_color_array.write[i] = random_variation_color;
-							line_color_array.write[i] = random_variation_color.lightened(0.2);
+							line_vertex_array.push_back(vertex1);
```

Comment:
You can probably resize above the loop for line_vertex_array and line_color_array with  2 * outline_size